File: flight-tracking-service/src/main/java/github/wozniak/flighttrackingservice/core/dto/PlaneDTO.java
package github.wozniak.flighttrackingservice.core.dto;

import github.wozniak.flighttrackingservice.core.entity.Plane;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
public class PlaneDTO {

    private String callSign;
    private String model;
    private int seatCapacity;
    private int luxurySeats;
    private int speedKnots;
    private int rangeMiles;

    public PlaneDTO(Plane plane){
        this.callSign = plane.getCallSign();
        this.model = plane.getModel();
        this.seatCapacity = plane.getSeatingCapacity();
        this.luxurySeats = plane.getLuxurySeats();
        this.speedKnots = plane.getCruisingSpeedKnots();
        this.rangeMiles = plane.getRangeMiles();
    }
}

File: backend/src/Domain/fleet/Dto/PlaneDto.cs
using backend.Domain.aircraft.Dto;
using backend.Domain.fleet.Entity;

namespace backend.Domain.fleet.Dto;

public class PlaneDto
{
    public string CallSign { get; set; }
    public AircraftDto Aircraft { get; set; }
    public string Status { get; set; }

    public PlaneDto(Plane plane)
    {
        CallSign = plane.CallSign;
        Aircraft = new AircraftDto(plane.Aircraft);
        Status = plane.Status.ToString();
    }
}